File: algorithms/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        results = dict()
        for index in range(len(nums)):
            if target - nums[index] in results:
                return [results[target - nums[index]], index]
            else:
                results[nums[index]] = index

File: algorithms/0001-two-sum.go
package algorithms

func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for i := 0; i < len(nums); i++ {
		if j, e := m[target -nums[i]]; e {
			return []int {j, i}
		} else {
			m[nums[i]] = i
		}
	}
	return nil
}
